### ISSUE TITLE
Is it possible to tune threshold for binary classifiers?

### ISSUE THREAD

**Issue by cjm715** [mlr-org/mlr3tuning#209]
.... or set the threshold to a fixed value in a tuning instance?

**Comment by berndbischl** [comment]
this is currently not supported directly in mlr3tuning, but definitely in scope as a next feature
@mllg @mb706  what do we currently have in mlr3 here and in mlr3pipelines?

**Comment by mllg** [comment]
There is another issue about this somewhere. We need to arrange a call to decide what to do with other predict-time hyperparameters.

**Comment by mllg** [comment]
There is is now a prototype to tune a single prediction in the threshold branch.

**Comment by berndbischl** [comment]
is it documented how to set thresholds simply with models? can you link it here for reference pls?
also my feeling is that threshold optimization should be a PipeOp in mlr3pipelines, so we must dicuss this with @mb706

**Comment by mb706** [comment]
This should be a `PipeOp` and would work relatively efficiently when we implement #212. There is a small problem here: This would not be as efficient as it could be if the `Learner` that produces the predictions has a slow `predict()` function. Ideally we would want some functionality that notices that only parameters of the `PipeOp` *after* the `Learner` have changed, and that evaluates the `PipeOp$predict()` multiple times, but the `Learner$train()` only once.

**Comment by pfistfl** [comment]
Started implementation in https://github.com/mlr-org/mlr3pipelines/pull/356.

**Comment by sebffischer** [comment]
this is implemented via `po("tunethreshold")`
